node_pool/main.tf

```diff
@@ -18,6 +18,9 @@ resource "random_id" "entropy" {
     name         = "${var.name}"
     region       = "${var.region}"
     disk_size    = "${var.disk_size_in_gb}"
+    tags         = "${join(",", sort(var.node_pool_tags))}"
+    disk_type    = "${var.disk_type}"
+    labels       = "${jsonencode(var.node_labels)}"
```

Comment:
After testing - @emailbob verified that this `jsonencode()` function will sort by the keys and keep consistent orders if the order changes.